### ISSUE TITLE
Not Fully Unsupervised as the Labels are used to compute the total loss 

### ISSUE THREAD

**Issue by smeenapadnekar** [ByungKwanLee/Causal-Unsupervised-Segmentation#12]
You can find the following line in `train_front_door_tr.py` [[Link](https://github.com/ByungKwanLee/Causal-Unsupervised-Segmentation/blob/48e6459d4f9f57ba732084d230a13c0ada70c6df/train_front_door_tr.py#L210C1-L213C67)]
```python
# Bank and EMA
cluster.bank_init()
ema_init(segment.head, segment.head_ema)
ema_init(segment.projection_head, segment.projection_head_ema
```

Here, the funcion of `ema_init` is defined in `modules/segment_module.py` [[Link](https://github.com/ByungKwanLee/Causal-Unsupervised-Segmentation/blob/48e6459d4f9f57ba732084d230a13c0ada70c6df/modules/segment_module.py#L342)]

```python
def ema_init(x, x_ema):
    for param, param_ema in zip(x.parameters(), x_ema.parameters()): param_ema.data = param.data; param_ema.requires_grad = False
```

_Originally posted by @ByungKwanLee in https://github.com/ByungKwanLee/Causal-Unsupervised-Segmentation/issues/5#issuecomment-1786983385_

**Comment by ChristophReich1996** [comment]
https://github.com/ByungKwanLee/Causal-Unsupervised-Segmentation/issues/5#issuecomment-2738221627
